### DISCUSSION TITLE
UQ SST divergence

### DISCUSSION THREAD

**Discussion by edaniele-tpi** [su2code/SU2#1152] [Q&A] [answered]
Dear all,

during the application of the UQ with the SST turbulence model on a 2D airfoil case I encounter solution divergence issues, in terms of continuously increasing force coefficients' amplifications over the iterations.
I follow the default UQ setup firstly, and later relax down to 0.05 to enhance stability, without success.
@jayantmukho : It is any further best practice suggested to mitigate this issue?

Thanks in advance for your helpful discussion.

Elia

**Comment by jayantmukho** [comment]
Hi Elia, would it be possible to get the config and geometry that you are using. I can have a closer look. 

Since the UQ module messes with the computations slightly, divergences can happen. But usually they are fixable. 

I am so very glad that you are using the UQ module :)

**Reply by edaniele-tpi** [reply]
Hi Jayant,

thanks for your quick reply!
I attached the configuration file for the case. Unfortunately I cannot share the grid file due to the geometry available there.
However, I add an the log file where summary information is provided as well, hoping it helps.

Thanks in advance for any suggestions and best regards,

Elia

[case.zip](https://github.com/su2code/SU2/files/5769114/case.zip)

**Reply by jayantmukho** [reply]
I checked the config file and made some basic updates that should help. I have uploaded it here and hopefully it achieves convergence. I am assuming that the baseline (non-UQ) simulation did converge. 

I haven't run the UQ module on an incompressible case, so I am interested in how this turns out! But is there a particular reason you are running incompressible? The Mach numbers is around 0.4, and there would be compressibility effects that would be lost in the incompressible solver. 

[case.zip](https://github.com/su2code/SU2/files/5773128/case.zip)

**Reply by edaniele-tpi** [reply]
Hi Jayant,

thanks for the proposed changes, I will test them right away.
Concerning your point on compressibility issue, I can say the incompressible simulation serves as baseline to measure the compressible one against. Thus, clearly, the case has to be repeated in a compressible scenario later on. Clearly, the baseline, input, incompressible solution is converged.
Keep you posted on the new results.

Best regards,

Elia

**Reply by edaniele-tpi** [reply]
Hi Jayant,

apparently, no big change in force coefficient history is obtained by changing the configuration file. 
I attached the scaled drag coefficient (w.r.t. 0.0100 as baseline) for both old and new configuration files.

![cd](https://user-images.githubusercontent.com/65940790/103866536-3baef300-50c6-11eb-8660-f50236a73f85.png)

Both cases started with (the same) fully converged incompressible solution. Any further guess?

**Reply by edaniele-tpi** [reply]
Hi Jayant,

I repeated the UQ variations based on a compressible case, and as foreseen by you, it did converge around a mean values with small oscillations.
Thanks again for your hints.
Best regards,

Elia

**Reply by jayantmukho** [reply]
Hey Elia!

I am glad it worked for the compressible case! I will include a warning about issues in using the UQ module with the incompressible solver. 

Thanks for reporting your progress with it. It's always nice to hear that module is working and being used. 

Cheers, 
Jayant

**Comment by edaniele-tpi** [comment, marked as answer]
Unfortunately the incompressible case lead to divergence. I switched to the related compressible version, performing the UQ tests with that.
